### PR TITLE
Add notion search

### PR DESCRIPTION
**Pull request by HenriChabrand**
## Description

Allow user to search Notion page.

## Type of change

- Bug fix / improvement for my extension

## Screencast

Screenshot 2021-11-18 at 09 26 48

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

File: extensions/notion/src/search-page.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  OpenInBrowserAction,
+  preferences,
+  showToast,
+  ToastStyle,
+  useNavigation,
+  setLocalStorageItem,
+  getLocalStorageItem,
+  getApplications,
+  getDefaultApplication,
+  closeMainWindow,
```
Comment (pernielsentikaer) [inline extensions/notion/src/search-page.tsx:21]:
```suggestion
  ActionPanel,
  Color,
  Icon,
  List,
  Detail,
  FormValues,
  preferences,
  showToast,
  ToastStyle,
  setLocalStorageItem,
  getLocalStorageItem,
  getApplications,
  closeMainWindow,
```

Comment (pernielsentikaer) [reply extensions/notion/src/search-page.tsx:21]: Remove unused imports

File: extensions/notion/src/search-page.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+} from '@raycast/api'
+import { useEffect, useState } from 'react'
+import {
+  Database,
+  DatabasePropertie,
+  Page,
+  fetchDatabases,
+  fetchDatabaseProperties,
+  createDatabasePage,
+  searchPages,
```
Comment (pernielsentikaer) [inline extensions/notion/src/search-page.tsx:31]:
```suggestion
  Page,
  searchPages,
```

Remove unused imports

File: extensions/notion/src/search-page.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        ))}
+      </List.Section>
+    </List>
+  ) 
+}
+
+export function NoSharedContent(): JSX.Element{
+  return (<Detail markdown={`## No Shared Content 
+  \n\n Make sure to **Invite** at least one database with the integration you have created.\n ![NotionShare](https://images.ctfassets.net/lzny33ho1g45/2pIkZOvLY2o2dwfnJIYJxt/d5d9f1318b2e79ad92d8197e4abab655/automate-notion-with-zapier-11-share-options.png)`} />)
+}
```
Comment (pernielsentikaer) [inline extensions/notion/src/search-page.tsx:187]:
```suggestion
```

Unused function, should it be used?

File: extensions/notion/src/search-page.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    'blue': Color.Blue,
+    'green': Color.Green,
+    'yellow': Color.Yellow,
+    'orange': Color.Orange,
+    'purple': Color.Purple,
+    'pink': Color.Magenta
+  } as Record<string,Color>
+
+  return colorMapper[notionColor] 
+}
```
Comment (pernielsentikaer) [inline extensions/notion/src/search-page.tsx:215]:
```suggestion
```

Unused function, should it be used?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @HenriChabrand 👋 

Awesome addition to the extension 💯 , while I'm waiting for a key to test up against I have a few suggestions that could be addressed in the meanwhile.

I'll do the last check later today when I have the right access to Notion 💣 

By the way, awesome extension and README 🚀

**Comment by HenriChabrand** [comment]
Hello @pernielsentikaer 👋 
Thanks for the review and suggestion – I've committed them all!

Let me know if there's anything I can do 🙌

**Comment by pernielsentikaer** [comment]
Hi @HenriChabrand 👋 

Thanks for the provided token in Slack, I'll test and get back to you 😄

**Comment by HenriChabrand** [comment]
Hello @pernielsentikaer 👋
Sorry, there were too many conflicts for some reason – so I've created another clean branch from master.

I've made a new pull request here:
https://github.com/raycast/extensions/pull/392

It simplifies the search and avoids this blinking effect + I found why Raycast was crashing (infinite hook loop).

**Pull request closed without merging**